### ISSUE TITLE
Feature Request: set directory for files

### ISSUE THREAD

**Issue by AlexPerathoner** [erik/derive#35]
Hi,
It'd be cool if you could set a directory with all the .gpx files I want to view, instead of dragging them every time to the window.
Maybe it's not even possible, but if it is I hope you'll implement this feature one day!

**Comment by erik** [comment]
As far as I know, there won't be a way for browsers to remember the files that are included without putting them all in local storage (which would likely be too much data) or uploading them to a server (which I do not want to do). 

What might be a bit of an improvement is supporting dragging directories as well as individual files. Seems like this is Chrome-only now.

**Comment by nck974** [comment]
Is there any change this has improved over the last years?

The performance of the app is great for the huge amount of data it plots. But dragging 6000 gpx files in a i7-8550U seems to not be possible. I have been able to drag in batches of 1000 with a 10-30sec freeze each time.

**Comment by erik** [comment]
Not really, this project hasn't been modified substantially since ~2018. It's honestly a testament to Leaflet's quality that it works well at all, it really is the simplest implementation you could imagine.

Maybe import could be sped up somewhat and made more responsive by moving parsing over to web workers.

**Comment by TylerHext** [comment]
I'm interested in this feature so I can retain my heatmap on a personal server that hosts the app on my local network, and references a directory of my GPX files.
Unfortunately I don't know JS, but this app is so perfect for the use case it feels silly (and time consuming) to rebuild it.

Any ideas on where one would start to convert from drag & drop to a static directory as source? Or a similar project that is closer to the desired functionality?

**Comment by hugovk** [comment]
> Or a similar project that is closer to the desired functionality?

Here's a couple of similar Python projects:

* https://github.com/sethoscope/heatmap
* https://github.com/marcusvolz/strava_py

And R:

* https://github.com/marcusvolz/strava

**Comment by erik** [comment]
Another one I've worked on recently:

https://github.com/erik/hotpot

**Comment by TylerHext** [comment]
> Another one I've worked on recently:
> 
> https://github.com/erik/hotpot

This is particularly encouraging. I'll check it out this weekend. Thanks!
